src/backend/env/mod.rs

```diff
@@ -274,6 +274,42 @@ pub enum Destination {
 }
 
 impl State {
+    pub fn delay_weekly_chores(&mut self, caller: Principal) -> bool {
+        let Some(user) = self
+            .principal_to_user(caller)
+            .and_then(|user| user.stalwart.then_some(user))
+        else {
+            return false;
+        };
+
+        // If we shifted already, exit.
+        if self.last_weekly_chores >= time() + WEEK {
+            return false;
+        }
+
+        self.weekly_chores_delay_votes.insert(user.id);
+
+        if self.weekly_chores_delay_votes.len() * 100
+            / self.users.values().filter(|user| user.stalwart).count()
+            >= CONFIG.report_confirmation_percentage as usize
+        {
+            self.last_weekly_chores += WEEK;
```

Comment:
Shouldn't we reset `self.weekly_chores_delay_votes`?

Also, I think with the current writing a stalwart could delay the weekly chores indefinitely as soon as the condition in L.292 is verified just by voting yes again.

Reply:
>Shouldn't we reset self.weekly_chores_delay_votes?

As answered above, since I assumed we will always need an upgrade and the field is marked as `skip`, an upgrade will automatically reset it. We can definitely reset the field on the next (delayed minting), but I am not sure if this is a good and realistic scenario. As I said, this operation is consider as a total out-of-order situation.

>Also, I think with the current writing a stalwart could delay the weekly chores indefinitely as soon as the condition in L.292 is verified just by voting yes again.

I don't follow. It's a hashmap, a repeated voting will be a noop. Am I missing something?

Reply:
Ok my bad, actually if we already shifted then we reject any new votes. Everything looks good, my comment about the test is not relevant.

Reply:
So each deploy will reset this field. We can also do it with own post upgrade function if needed.